cognite/powerops/client/shop/data_classes/dayahead_trigger.py

```diff
@@ -53,19 +133,28 @@ class Case(BaseModel):
 
     case_name: str  # e.g. Glomma
     pre_runs: list[ShopRun]
-    plants: list[str] = []
-    commands_file: Optional[str] = None  # most prerun files do not have commands. Provide this as extra files
+    commands_file: Optional[str] = None
     pre_runs_external_id_prefix: Optional[str] = None
 
 
 class DayaheadTrigger(BaseModel):
+    """
+    Contains the blueprint of a Dayahead workflow definition.
+    This object is used together with the DayaheadTriggerAPI to trigger a workflow from an instance of this class.
+    """
+
     price_scenarios: list[str]
     main_scenario: str = ""
     shop_version: Optional[str] = _SHOP_VERSION_FALLBACK
     price_area: str
     method: str
     cases: list[Case]
+    _plants_per_workflow: list[str]
 
     @property
-    def num_price_areas(self) -> int:
-        return len(self.price_scenarios)
+    def plants_per_workflow(self):
+        return self._plants_per_workflow
+
+    @plants_per_workflow.setter
+    def plants_per_workflow(self, value):
+        self._plants_per_workflow = value
```

Comment:
Any reason why this is private?

Reply:
Because the plants are read from CDF from the pre run files. I was in doubt whether to add it as a config field as well in case there is a conflict between the retrieved plants and the specified plants by the user. And a validation would not discover this until all the shop files from cdf have been processed. I guess I could add a validation that fetches all shop files and validates the plants before triggering the workflow though? although it would entail more calls to cdf, and a slightly slower workflow

Reply:
Then, I wonder why you let the user set plants, if that is given by the pre run files? Is it possible to allow the user to only set it one place, and thus avoiding any potential conflict?